cmd/deploy_test.go

```diff
@@ -1795,103 +1795,103 @@ func TestReDeploy_OnRegistryChange(t *testing.T) {
-
-	// First: prebuild Function
-	if err := cmdBuild.Execute(); err != nil {
+	if err := f.Write(); err != nil {
 		t.Fatal(err)
 	}
 
-	// change registry and deploy again
-	newRegistry := "example.com/fred"
-
-	cmd := NewDeployCmd(NewTestClient(
-		fn.WithDeployer(mock.NewDeployer()),
-	))
-
-	cmd.SetArgs([]string{"--registry=" + newRegistry, "--build=false"})
-
-	// Second: Deploy with different registry and expect 'not built' error because
-	// registry has changed but build is disabled
+	// Deploying again with a new registry should trigger "not built" error
+	// if specifying --build=false
+	cmd := NewDeployCmd(NewTestClient())
+	cmd.SetArgs([]string{"--registry", registry, "--build=false"})
 	if err := cmd.Execute(); err == nil {
-		t.Fatal(err)
+		// TODO: should be a typed error which can be checked.  This will
+		// succeed even if an unrelated error is thrown.
+		t.Fatalf("expected error 'not built' not received")
 	}
 
-	// ASSERT
-	expectF, err := fn.NewFunction(root)
-	if err != nil {
-		t.Fatal("couldnt load function from path")
+	// Assert that the requested change did not take effect
+	if f, err = fn.NewFunction(root); err != nil {
+		t.Fatal(err)
 	}
-
-	if !strings.Contains(expectF.Build.Image, TestRegistry) {
+	if strings.Contains(f.Build.Image, registry) {
 		t.Fatal("expected registry to NOT change since --build=false")
 	}
 }
 
-// TestDeploy_NoErrorOnOldFunctionNotFound assures that no error is given when old Function's
-// service is not available (is already deleted manually or the namespace doesnt exist etc.)
+// TestDeploy_NoErrorOnOldFunctionNotFound assures that no error is given when
+// old Function's service is not available (is already deleted manually or the
+// namespace doesnt exist etc.)
 func TestDeploy_NoErrorOnOldFunctionNotFound(t *testing.T) {
 	var (
-		root    = fromTempDirectory(t)
+		root    = FromTempDirectory(t)
 		nsOne   = "nsone"
 		nsTwo   = "nstwo"
 		remover = mock.NewRemover()
 	)
 
-	// Simulate remover error
+	// A remover which can not find the old instance
 	remover.RemoveFn = func(n, ns string) error {
-		return apiErrors.NewNotFound(schema.GroupResource{Group: "", Resource: "Namespace"}, nsOne)
+		// Note that the knative remover explicitly checks for
+		// if it received an apiErrors.IsNotFound(err) and if so returns
+		// a fn.ErrFunctionNotFound.  This test implementation is dependent
+		// on that.  This is a change from the original implementation which
+		// directly returned a knative erorr with:
+		//   return apiErrors.NewNotFound(schema.GroupResource{Group: "", Resource: "Namespace"}, nsOne)
+		if ns == nsOne {
+			// Fabricate a not-found error.  For example if the function
+			// or its namespace had been manually removed
+			return fn.ErrFunctionNotFound
+		}
+		return nil
 	}
+	clientFn := NewTestClient(
+		fn.WithDeployer(mock.NewDeployer()),
+		fn.WithRemover(remover),
+		fn.WithPipelinesProvider(mock.NewPipelinesProvider()),
+	)
```

Comment:
is the pipelines provider necessary here?

Reply:
Good call.  No; that should not be necessary.  Let's give it a try with that removed...